gobblin-data-management/src/test/java/org/apache/gobblin/data/management/copy/iceberg/IcebergPartitionDatasetTest.java

```diff
@@ -0,0 +1,323 @@
+    srcFilePaths.add(SRC_WRITE_LOCATION + "/file1.orc");
+    srcFilePaths.add(SRC_WRITE_LOCATION + "/file2.orc");
+    srcFilePaths.add(SRC_WRITE_LOCATION + "/file3.orc");
+    srcFilePaths.add(SRC_WRITE_LOCATION + "/file4.orc");
+    srcFilePaths.add(SRC_WRITE_LOCATION + "/file5.orc");
+
+    List<DataFile> srcDataFiles = getDataFiles();
+    Mockito.when(srcIcebergTable.getPartitionSpecificDataFiles(Mockito.any())).thenReturn(srcDataFiles);
+
+    icebergPartitionDataset = new TestIcebergPartitionDataset(srcIcebergTable, destIcebergTable, properties, sourceFs,
+        true);
+
+    CopyConfiguration copyConfiguration =
+        CopyConfiguration.builder(targetFs, copyConfigProperties).preserve(PreserveAttributes.fromMnemonicString(""))
+            .copyContext(new CopyContext()).build();
+
+    Collection<CopyEntity> copyEntities = icebergPartitionDataset.generateCopyEntities(targetFs, copyConfiguration);
+
+    Assert.assertEquals(copyEntities.size(), 6);
+    verifyCopyEntities(copyEntities, true);
+  }
+
+  @Test
+  public void testWithDifferentSrcAndDestTableWriteLocation() throws IOException {
+    srcFilePaths.add(SRC_WRITE_LOCATION + "/randomFile--Name.orc");
+    Mockito.when(srcTableMetadata.property(TableProperties.WRITE_DATA_LOCATION, "")).thenReturn(SRC_WRITE_LOCATION);
+    Mockito.when(destTableMetadata.property(TableProperties.WRITE_DATA_LOCATION, "")).thenReturn(DEST_WRITE_LOCATION);
+
+    List<DataFile> srcDataFiles = getDataFiles();
+    Mockito.when(srcIcebergTable.getPartitionSpecificDataFiles(Mockito.any())).thenReturn(srcDataFiles);
+
+    icebergPartitionDataset = new TestIcebergPartitionDataset(srcIcebergTable, destIcebergTable, properties, sourceFs,
+        true);
+
+    CopyConfiguration copyConfiguration =
+        CopyConfiguration.builder(targetFs, copyConfigProperties).preserve(PreserveAttributes.fromMnemonicString(""))
+            .copyContext(new CopyContext()).build();
+
+    List<CopyEntity> copyEntities =
+        (List<CopyEntity>) icebergPartitionDataset.generateCopyEntities(targetFs, copyConfiguration);
+
+    Assert.assertEquals(copyEntities.size(), 2);
+    verifyCopyEntities(copyEntities, false);
+  }
+
+  private List<DataFile> getDataFiles() throws IOException {
+    List<DataFile> dataFiles = new ArrayList<>();
+    for (String srcFilePath : srcFilePaths) {
+      DataFile dataFile = Mockito.mock(DataFile.class);
+      Path dataFilePath = new Path(srcFilePath);
+      Path qualifiedPath = sourceFs.makeQualified(dataFilePath);
+      Mockito.when(dataFile.path()).thenReturn(dataFilePath.toString());
+      Mockito.when(sourceFs.getFileStatus(Mockito.eq(dataFilePath))).thenReturn(getFileStatus(qualifiedPath));
+      dataFiles.add(dataFile);
+    }
+    return dataFiles;
+  }
+
+  private static FileStatus getFileStatus(Path path) {
+    FileStatus fileStatus = new FileStatus();
+    fileStatus.setPath(path);
+    return fileStatus;
+  }
```

Comment:
the name `get` is very confusing to associate w/ `setPath`.

actually, here it looks like you `createEmptyFileStatus(Path path)`